client/ayon_traypublisher/plugins/create/create_csv_ingest.py

```diff
@@ -835,6 +895,23 @@ def _create_instances_from_csv_data(self, csv_dir: str, filename: str):
                 "prepared_data_for_repres": []
             }
 
+            if product_item.has_promised_context:
+                hierarchy, _ = folder_path.rsplit("/",1)
+                families.append("csv_ingest_shot")
+                instance_data.update(
+                    {
+                        "newHierarchyIntegration": True,
+                        "hierarchy": hierarchy,
+                        "parents": product_item.parents
+                    }
+                )
+                # TODO create new task from provided task name
+#                if product_item.task_name:
+#                    instance_data["tasks"] = {
+#                        "name": product_item.task_name,
+#                        "type": "Generic"
+#                    }
```

Comment:
I wasn't sure what the `tasks` data should be here in order for `Extract Hierarchy to AYON` to create a new task along the new shots. From other code I thought the `dict` was correct but it looks like it's not.

![traypublisher_task_dict](https://github.com/user-attachments/assets/1fdf8423-cb67-44b2-936b-2bc247233132)

Unless it's an easy one and I missed something, I think this needs to be fixed in its own PR.

Reply:
I believe the tasks data should be part of the instances. But guess it isn't there as somehow the commented code is a bit hard-coded and it is not updating the dictionary data with name and type per task.

Reply:
I'm confused now because if I put in a list, it fails here: https://github.com/ynput/ayon-core/blob/6a07de6eb904c139f6d346fd6f2a7d5042274c71/client/ayon_core/plugins/publish/collect_anatomy_instance_data.py#L497
```
AttributeError: 'list' object has no attribute 'get'
```

Reply:
I think it is task info by name
```
"tasks": {
    "Modeling": {
        "type": "modeling",
    },
    "Animation": {
        "type": "animation",
    },
    ...
}
```

Reply:
That was it, thanks a lot for that !